### PR TITLE
[SLBeta2] Add Bigquery data transfer connector

### PR DESCRIPTION
**Pull request by SkNuwanTissera**
## Purpose
Add Bigquery data transfer connector
resolves https://github.com/wso2-enterprise/choreo/issues/7684

### CODE REVIEW

File: openapi/bigquery.datatransfer/client.bal [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+    # + uploadProtocol - Upload protocol for media (e.g. "raw", "multipart"). 
+    # + uploadType - Legacy upload protocol for media (e.g. "media", "multipart"). 
+    # + return - Successful response 
+    remote isolated function deleteTransferRun(string name, string? xgafv = (), string? accessToken = (), string? alt = (), string? callback = (), string? fields = (), string? 'key = (), string? oauthToken = (), boolean? prettyPrint = (), string? quotaUser = (), string? uploadProtocol = (), string? uploadType = ()) returns Empty|error {
+        string  path = string `/v1/${name}`;
+        map<anydata> queryParam = {"$.xgafv": xgafv, "access_token": accessToken, "alt": alt, "callback": callback, "fields": fields, "key": 'key, "oauth_token": oauthToken, "prettyPrint": prettyPrint, "quotaUser": quotaUser, "upload_protocol": uploadProtocol, "uploadType": uploadType};
+        path = path + check getPathForQueryParam(queryParam);
+        http:Request request = new;
+        //TODO: Update the request as needed;
+        Empty response = check self.clientEp-> delete(path, request, targetType = Empty);
```
Comment (LakshanSS) [inline openapi/bigquery.datatransfer/client.bal:113]: For these kind of operations, we remove the Empty in the openapi.yml

This is common for the google connectors.

File: openapi/bigquery.datatransfer/openapi.yml [hunk 2]
```diff
@@ -0,0 +1,1280 @@
...
+      description: Represents preferences for sending email notifications for transfer run events.
+      properties:
+        enableFailureEmail:
+          description: If true, email notifications will be sent on transfer run failures.
+          type: boolean
+      type: object
+    Empty:
+      description: "A generic empty message that you can re-use to avoid defining duplicated empty messages in your APIs. A typical example is to use it as the request or the response type of an API method. For instance: service Foo { rpc Bar(google.protobuf.Empty) returns (google.protobuf.Empty); } The JSON representation for `Empty` is empty JSON object `{}`."
+      properties: {}
+      type: object
```
Comment (LakshanSS) [inline openapi/bigquery.datatransfer/openapi.yml:921]: This thing has to be removed

File: openapi/bigquery.datatransfer/openapi.yml [hunk 3]
```diff
@@ -0,0 +1,1280 @@
...
+          name: name
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Empty"
```
Comment (LakshanSS) [inline openapi/bigquery.datatransfer/openapi.yml:60]: This has to be removed

File: openapi/bigquery.datatransfer/client.bal (hunk 1 above)
Comment (SkNuwanTissera) [reply openapi/bigquery.datatransfer/client.bal]: Changed as suggested. 
/Thanks

File: openapi/bigquery.datatransfer/openapi.yml (hunk 3 above)
Comment (SkNuwanTissera) [reply openapi/bigquery.datatransfer/openapi.yml]: Changed as suggested. 
/Thanks

File: openapi/bigquery.datatransfer/openapi.yml (hunk 2 above)
Comment (SkNuwanTissera) [reply openapi/bigquery.datatransfer/openapi.yml]: Changed as suggested. 
/Thanks

File: openapi/bigquery.datatransfer/Module.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+### Step 1: Import connector
+First, import the `ballerinax/googleapis.bigquery.datatransfer` module into the Ballerina project.
+```ballerina
+    import ballerinax/googleapis.bigquery.datatransfer;
+```
+
+### Step 2: Create a new connector instance
+Create a `datatransfer:ClientConfig` with the OAuth2 tokens obtained, and initialize the connector with it. 
+```ballerina
+    datatransfer:ClientConfig clientConfig = {authConfig : {token: "<Your access token>"}};
```
Comment (indikasampath2000) [inline openapi/bigquery.datatransfer/Module.md:25]: `authConfig` must be `auth`

Comment (SkNuwanTissera) [reply openapi/bigquery.datatransfer/Module.md]: Changed as suggested. /Thanks

**Review by indikasampath2000 — APPROVED** [review]

**Pull request merged**